tasks/base.py

```diff
@@ -114,20 +123,33 @@ def on_timeout(self, soft: bool, timeout: int):
 class BaseCodecovTask(celery_app.Task):
     Request = BaseCodecovRequest
 
-    def __init_subclass__(cls, name=None):
+    def __init_subclass__(cls, name="unknown_task"):
         cls.name = name
+        # All task names follow the format `app.[cron|task].<task_group>.<task>`
+        task_group = name.split(".")[-2] if name != "unknown_task" else "unknown_group"
```

Comment:
does this give an error if the length of `name.split('.')` is 1?

Reply:
Yes, but all tasks are supposed to have a `task_group` in their name (separated by `'.'`. Check [the celery config](https://github.com/codecov/shared/blob/main/shared/celery_config.py)